### ISSUE TITLE
Set minimum gcsfs version

### ISSUE THREAD

**Issue by jacobtomlinson** [nteract/papermill#342]
It looks like there is a problem with using papermill 0.19.1 with gcsfs 0.1.1. Due to unpinned versions it is possible to end up in this state.

It would be good to update the dependencies to require `gcsfs>=0.2.1`.

## Minimum reproducable example

```
$ conda create -n papermill python=3.6 dask==1.1.5 papermill==0.19.1 gcsfs==0.1.1 -y

$ conda activate papermill

(papermill) $ python -c 'import papermill'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/jovyan/.local/share/conda/envs/papermill/lib/python3.6/site-packages/papermill/__init__.py", line 5, in <module>
    from .api import display, record, read_notebook, read_notebooks
  File "/home/jovyan/.local/share/conda/envs/papermill/lib/python3.6/site-packages/papermill/api.py", line 17, in <module>
    from .iorw import load_notebook_node, list_notebook_files
  File "/home/jovyan/.local/share/conda/envs/papermill/lib/python3.6/site-packages/papermill/iorw.py", line 47, in <module>
    from gcsfs import GCSFileSystem
  File "/home/jovyan/.local/share/conda/envs/papermill/lib/python3.6/site-packages/gcsfs/__init__.py", line 4, in <module>
    from .dask_link import register as register_dask
  File "/home/jovyan/.local/share/conda/envs/papermill/lib/python3.6/site-packages/gcsfs/dask_link.py", line 56, in <module>
    register()
  File "/home/jovyan/.local/share/conda/envs/papermill/lib/python3.6/site-packages/gcsfs/dask_link.py", line 16, in register
    class DaskGCSFileSystem(GCSFileSystem, dask.bytes.core.FileSystem):
AttributeError: module 'dask.bytes.core' has no attribute 'FileSystem'
```

**Comment by MSeal** [comment]
Likely this is an incompatibility with using an old gcsfs and recent dask. I am guessing an older dask version from around the release time of 0.1.1 would work. We don't really guard users from picking non-recent versions in bad combinations unless they directly conflict with the library's calls. Papermill itself can talk to 0.1 or 0.2 syntax but I see where you can easily get into trouble if you are only pinning gcsfs back.

The `No module named 'dask'` is purely a gcsfs 0.1 message. I tested it out and 0.2 doesn't emit this warning when dask is missing.

@gogasca Would there be complications with pinning to `>=0.2.1` or `>=0.2.0` and how should we think of users of this library? Is it generally recommended to upgrade to latest or do many people still use `0.1`?

**Comment by MSeal** [comment]
I'm going to set the requirement to >=0.2.0 in papermill 1.0 release today. It seems reasonable to suggest users be at least to that requirement. Folks can push back if we need to make a 1.0.1 release with less strict requirements.

**Comment by gogasca** [comment]
[gcsfs 0.2.1](https://pypi.org/project/gcsfs/#history) implemented the [exception](https://github.com/dask/gcsfs/pull/131) that enables the retry logic for writing cells [very fast in GCS](https://cloud.google.com/storage/docs/key-terms#immutability)
Read operation from notebooks stored in GCS via papermill would work in gcsfs in any version including 0.1.x, but write functionality (without issues) won't be available unless you are using >=0.2.1

**Comment by MSeal** [comment]
Pinned with 1.0 to a later version to help with write issues.
